packages/ai-chat/src/browser/context-file-validation-service.spec.ts

```diff
@@ -0,0 +1,392 @@
+        });
+
+        it('should reject existing file with file:// URI in /tmp', async () => {
+            const result = await validationService.validateFile('file:///tmp/temporary-file.log');
+            expect(result).to.be.false;
+        });
+
+        it('should reject non-existing file with file:// URI outside workspace', async () => {
+            const result = await validationService.validateFile('file:///var/log/nonexistent.log');
+            expect(result).to.be.false;
+        });
+
+        it('should validate file at workspace root with file:// URI', async () => {
+            const result = await validationService.validateFile('file:///home/user/workspace/package.json');
+            expect(result).to.be.true;
+        });
+    });
+
+    describe('validateFile with URI objects', () => {
+        it('should validate existing file with URI object within workspace', async () => {
+            const uri = new URI('file:///home/user/workspace/src/index.tsx');
+            const result = await validationService.validateFile(uri);
+            expect(result).to.be.true;
+        });
+
+        it('should reject non-existing file with URI object within workspace', async () => {
+            const uri = new URI('file:///home/user/workspace/src/missing.tsx');
+            const result = await validationService.validateFile(uri);
+            expect(result).to.be.false;
+        });
+
+        it('should reject existing file with URI object outside workspace', async () => {
+            const uri = new URI('file:///etc/passwd');
+            const result = await validationService.validateFile(uri);
+            expect(result).to.be.false;
+        });
+
+        it('should reject another existing file with URI object outside workspace', async () => {
+            const uri = new URI('file:///home/other-user/secret.txt');
+            const result = await validationService.validateFile(uri);
+            expect(result).to.be.false;
+        });
+    });
+
+    describe('validateFile with no workspace', () => {
+        beforeEach(() => {
+            // Override mock to return no workspace roots
+            mockWorkspaceService.tryGetRoots = () => [];
+        });
+
+        it('should reject any file when no workspace is open', async () => {
+            const result = await validationService.validateFile('src/index.tsx');
+            expect(result).to.be.false;
+        });
+
+        it('should reject absolute path when no workspace is open', async () => {
+            const result = await validationService.validateFile('/home/user/file.txt');
+            expect(result).to.be.false;
+        });
+
+        it('should reject file:// URI when no workspace is open', async () => {
+            const result = await validationService.validateFile('file:///home/user/file.txt');
+            expect(result).to.be.false;
+        });
+    });
+
+    describe('validateFile with multiple workspace roots', () => {
```

Comment:
This is tricky: I believe that our agents are still restricted to operating in the [first workspace root](https://github.com/eclipse-theia/theia/blob/8dd3f8054344e74260eb6cbbb18dca86589ef018/packages/ai-ide/src/browser/workspace-functions.ts#L53-L59), so I'm not sure it makes sense to indicate that files in other roots are ok.